packages/shared/graphql/GraphQLClient.ts

```diff
@@ -40,15 +40,19 @@ export class GraphQLClient implements GraphQLClientInterface {
       data.query = data.query.loc?.source.body ?? "";
     }
 
+    const headers: Record<string, string> = {
+      "Content-Type": "application/json",
+    };
+    if (accessToken) {
+      headers["Authorization"] = `Bearer ${accessToken}`;
+    }
+    if (testScope) {
+      headers["replay-test-scope"] = testScope;
+    }
+
     const response = await fetch(this.url, {
       method: "POST",
-      headers: {
-        ...(accessToken && {
-          Authorization: `Bearer ${accessToken}`,
-          "replay-test-scope": this.testScope,
```

Comment:
I would expect this to send the header `"replay-test-scope": undefined` but you say you're seeing `"replay-test-scope": "undefined"`?

Reply:
That's what I had expected as well (more precisely I expected headers with `undefined` values to not be sent - you can't set an HTTP header to `undefined`). I guess that `fetch()` coerces all header values to `string`.

Reply:
Interesting. I had tested it in Chrome and it shows the header being sent as `undefined` (not a string). But looking again, I see that Chrome's Network panel only shows quotes for _certain_ strings and hides them for others. That's tricky :)